### ISSUE TITLE
Material option to enable anisotropic texture filtering

### ISSUE THREAD

**Issue by aglitchman** [defold/defold#6155]
**Is your feature request related to a problem? Please describe (REQUIRED):**
Now, we can choose only between Bilinear (Linear Mipmap Nearest) or Trilinear (Linear Mipmap Linear) filtering. 

They look ugly if you are going to develop a beautiful 3D game.

**Describe the solution you'd like (REQUIRED):**
I would like to have an option to set anisotropic filtering for textures per material basis. 

For WebGL, for example, we can set it via the `EXT_texture_filter_anisotropic` extension.

**Describe alternatives you've considered (REQUIRED):**
I use this hack for the HTML5 platform to enable anisotropic filtering for all textures:

```html
<script>
// This enables anisotropic filtering on mipmapped textures
[window.WebGLRenderingContext, window.WebGL2RenderingContext].forEach(function (context) {
    if (!context) return;

    var maxAnisotropy = 2; // <-- Change this

    var extAnisotropic = false;
    var texParameteriOld = context.prototype.texParameteri;
    context.prototype.texParameteri = function (target, pname, param) {
        if (
            target == this.TEXTURE_2D &&
            pname == this.TEXTURE_MIN_FILTER &&
            (param == this.LINEAR_MIPMAP_LINEAR || param == this.LINEAR_MIPMAP_NEAREST)
        ) {
            if (!extAnisotropic) {
                // Initialize it once
                var ext =
                    this.getExtension("EXT_texture_filter_anisotropic") ||
                    this.getExtension("MOZ_EXT_texture_filter_anisotropic") ||
                    this.getExtension("WEBKIT_EXT_texture_filter_anisotropic");
                var max = ext ? this.getParameter(ext.MAX_TEXTURE_MAX_ANISOTROPY_EXT) : 0;
                extAnisotropic = {
                    ext: ext,
                    max: Math.min(maxAnisotropy, max),
                };
            }

            if (extAnisotropic.ext) {
                // Set it
                this.texParameterf(
                    this.TEXTURE_2D,
                    extAnisotropic.ext.TEXTURE_MAX_ANISOTROPY_EXT,
                    extAnisotropic.max
                );
            }
        }

        texParameteriOld.call(this, target, pname, param);
    };
});
</script>
```

There is no way to do that on all other platforms.

**Additional context (OPTIONAL):**

[Arm's documentation](https://developer.arm.com/documentation/101897/0200/buffers-and-textures/anisotropic-sampling-performance) states that it's viable to use 2x bilinear AF in preference to isotropic trilinear filtering:

> ### How to optimize anisotropic filtering
> Try using the following optimization steps:
> 
> * Start using a max anisotropy of two, and then assess if it provides sufficient quality. Higher numbers of samples can improve quality, but they also give diminishing returns which are often not worth the performance cost.
> * **Consider using 2x bilinear AF in preference to isotropic trilinear filtering.** 2x bilinear is faster, and has better image quality in regions of high anisotropy. Note, that by switching to bilinear filtering, you can see some visible seams at the decision point between mipmap levels.
> * Only use AF and trilinear filtering for objects that benefit from it the most.

**Comment by Jhonnyg** [comment]
I think maybe we’d add it as a sampler property on the material. It’s not universally available on all devices but we can always detect that and just skip it if that’s the case. I can make a design doc for this 👍

**Comment by Jhonnyg** [comment]
Fixed by #6873
